src/Lecture4.hs

```diff
@@ -134,8 +140,21 @@ errors. We will simply return an optional result here.
 🕯 HINT: Use the 'readMaybe' function from the 'Text.Read' module.
 -}
 
+split :: Eq a => a -> [a] -> [[a]]
+split sep list = case elemIndex sep list of
+    Just i -> let (xs, ys) = splitAt i list
```

Comment:
`elemIndex` and `splitAt` traverse the list prefix twice. You can use standard functions `span` or `break` to split the list by the first occurrence of delimiter in a single traversal ➡️ 1️⃣